### PR TITLE
Collection Create and Add to favorite

### PR DESCRIPTION
**Pull request by Abhinav-grd**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@
 /node_modules
 /.pnp
 .pnp.js
+package-lock.json
```
Comment (pushkar8723) [inline .gitignore:7]: we should not ignore `package-lock.json` as it helps maintain consistency between dev and prod systems.

File: src/pages/gallery/components/Collections.tsx [hunk 2]
```diff
@@ -57,6 +57,7 @@ export default function Collections(props: CollectionProps) {
         <Wrapper>
             <Chip active={!selected} onClick={clickHandler()}>All</Chip>
             {collections?.map(item => <Chip
+                key={item.id}
```
Comment (pushkar8723) [inline src/pages/gallery/components/Collections.tsx:60]: 👍🏼

File: src/pages/gallery/components/CreateCollection.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    const { acceptedFiles, setProgressView, progressBarProps, refetchData, modalView, closeModal, closeUploadModal } = props;
+    const [albumName, setAlbumName] = useState("");
+
+    const handleChange = (event) => { setAlbumName(event.target.value); }
+
+    useEffect(() => {
+        if (acceptedFiles == null)
+            return;
+        let commonPathPrefix: string = (() => {
+            const A: string[] = acceptedFiles.map(files => files.path);
```
Comment (pushkar8723) [inline src/pages/gallery/components/CreateCollection.tsx:19]: We should avoid 1 char variable names.

File: src/utils/common/apiUtil.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
 export const getEndpoint = () => {
-    return process.env.NEXT_PUBLIC_ENTE_ENDPOINT || "https://api.ente.io";
+    const endPoint = process.env.NEXT_PUBLIC_ENTE_ENDPOINT ?? "https://api.ente.io";
+    console.log(endPoint);
```
Comment (pushkar8723) [inline src/utils/common/apiUtil.ts:3]: Avoid  `console.log` these will reflect in production logs as well and are also not a very efficient way of debugging. 
I will add a lint check, which will get rid of this.

File: .gitignore (hunk 1 above)
Comment (Abhinav-grd) [reply .gitignore:7]: image

you told to do this na?

Comment (pushkar8723) [reply .gitignore]: i meant just delete the file. Not add it in git ignore.

**Review by pushkar8723 — APPROVED** [review]

**Pull request merged**